driver/src/main/java/org/neo4j/driver/internal/metrics/MicrometerConnectionPoolMetrics.java

```diff
@@ -0,0 +1,277 @@
+    {
+        acquiring.decrementAndGet();
+    }
+
+    @Override
+    public void afterAcquiredOrCreated( ListenerEvent<?> acquireEvent )
+    {
+        acquired.increment();
+        Timer.Sample sample = ((MicrometerTimerListenerEvent) acquireEvent).getSample();
+        sample.stop( totalAcquisitionTimer );
+    }
+
+    @Override
+    public void afterTimedOutToAcquireOrCreate()
+    {
+        timedOutToAcquire.increment();
+    }
+
+    @Override
+    public void acquired( ListenerEvent<?> inUseEvent )
+    {
+        inUseEvent.start();
+    }
+
+    @Override
+    public void released( ListenerEvent<?> inUseEvent )
+    {
+        released.increment();
+        Timer.Sample sample = ((MicrometerTimerListenerEvent) inUseEvent).getSample();
+        sample.stop( totalInUseTimer );
+    }
+
+    @Override
+    public String id()
+    {
+        return this.id;
+    }
+
+    @Override
+    public int inUse()
+    {
+        return inUseSupplier.getAsInt();
+    }
+
+    @Override
+    public int idle()
+    {
+        return idleSupplier.getAsInt();
+    }
+
+    @Override
+    public int creating()
+    {
+        return creating.get();
+    }
+
+    @Override
+    public long created()
+    {
+        return toLong( created );
+    }
+
+    @Override
+    public long failedToCreate()
+    {
+        return toLong( failedToCreate );
+    }
+
+    @Override
+    public long closed()
+    {
+        return toLong( closed );
+    }
+
+    @Override
+    public int acquiring()
+    {
+        return acquiring.get();
+    }
+
+    @Override
+    public long acquired()
+    {
+        return toLong( acquired );
+    }
+
+    @Override
+    public long timedOutToAcquire()
+    {
+        return toLong( timedOutToAcquire );
+    }
+
+    @Override
+    public long totalAcquisitionTime()
+    {
+        return (long) totalAcquisitionTimer.totalTime( TimeUnit.MILLISECONDS );
+    }
+
+    @Override
+    public long totalConnectionTime()
+    {
+        return (long) totalConnectionTimer.totalTime( TimeUnit.MILLISECONDS );
+    }
+
+    @Override
+    public long totalInUseTime()
+    {
+        return (long) totalInUseTimer.totalTime( TimeUnit.MILLISECONDS );
+    }
+
+    @Override
+    public long totalInUseCount()
+    {
+        return toLong( released );
+    }
+
+    @Override
+    public String toString()
+    {
+        return format( "%s=[created=%s, closed=%s, creating=%s, failedToCreate=%s, acquiring=%s, acquired=%s, " +
+                       "timedOutToAcquire=%s, inUse=%s, idle=%s, " +
+                       "totalAcquisitionTime=%s, totalConnectionTime=%s, totalInUseTime=%s, totalInUseCount=%s]",
+                       id(), created(), closed(), creating(), failedToCreate(), acquiring(), acquired(),
+                       timedOutToAcquire(), inUse(), idle(),
+                       totalAcquisitionTime(), totalConnectionTime(), totalInUseTime(), totalInUseCount() );
+    }
+
+    private long toLong( Counter counter )
```

Comment:
```suggestion
    private long count( Counter counter )
```
or something similar, more "business logic" oriented?